src/plugins/data/public/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -302,15 +302,15 @@ export class IndexPattern implements IIndexPattern {
       return;
     }
     field.count = count;
-    await this.save();
+    return this.save();
   }
 
   getNonScriptedFields() {
-    return _.where(this.fields, { scripted: false });
+    return _.where(this.fields, { scripted: false }) as IFieldList;
```

Comment:
Yeah great question, it's because it references the generic of the first argument as `Array<T>` to assume the return value as `T[]` thus `IFieldType[]` and not `IFeildList`.

I now realize this likely does not include the added methods on `IFieldList`, might be worth putting this method on the `IFieldList` class.

```ts
where<T, U extends {}>(list: Array<T>, properties: U): T[];
```

Reply:
added method to `FieldList` here d143c22